test/node/download-metadata.js

```diff
@@ -0,0 +1,65 @@
+var fixtures = require('webtorrent-fixtures')
+var http = require('http')
+var test = require('tape')
+var WebTorrent = require('../../')
+
+function createServer (data, cb) {
+  var server = http.createServer(function (req, res) {
+    res.end(data)
+  }).listen(function () {
+    var address = server.address()
+    if (address.family === 'IPv6') {
+      address.address = '[' + address.address + ']'
+    }
+    var url = 'http://' + address.address + ':' + address.port + '/'
+    cb(url, next)
+  })
+
+  function next () {
+    server.close()
+  }
+}
+
+test('Download metadata for magnet URI with xs parameter', function (t) {
+  t.plan(2)
+
+  var client = new WebTorrent({ dht: false, tracker: false })
+
+  createServer(fixtures.leaves.torrent, function (url, next) {
+    client.add(fixtures.leaves.magnetURI + '&xs=' + encodeURIComponent(url), function (torrent) {
+      t.equal(torrent.name, 'Leaves of Grass by Walt Whitman.epub')
```

Comment:
The name is also in the magnet URI as the `&dn` parameter. Might be better to check for another parameter.